sys/include/net/gnrc/netif/hdr.h

```diff
@@ -201,20 +201,45 @@ void gnrc_netif_hdr_print(gnrc_netif_hdr_t *hdr);
 
 /* @brief   Check if packet is for broadcast
  *
- * @param[in]   pkt     packet to check
+ * @param[in]   hdr     gnrc_netif header to check
  *
  * @return              true if the packet is for broadcast, otherwise false
  */
-bool gnrc_netif_hdr_chk_pkt_bcast(gnrc_pktsnip_t* pkt);
+static inline bool gnrc_netif_hdr_chk_pkt_bcast(gnrc_netif_hdr_t *hdr)
+{
+    return ((hdr == NULL) ? false :
+            (hdr->flags & GNRC_NETIF_HDR_FLAGS_BROADCAST));
+}
+
+/* @brief   Check if packet is for multicast
+ *
+ * @param[in]   hdr     gnrc_netif header to check
+ *
+ * @return              true if the packet is for multicast, otherwise false
+ */
+static inline bool gnrc_netif_hdr_chk_pkt_mcast(gnrc_netif_hdr_t *hdr)
+{
+    return ((hdr == NULL) ? false :
+            (hdr->flags & GNRC_NETIF_HDR_FLAGS_MULTICAST));
+}
+
+/* @brief   Extract the destination address out of a gnrc_netif header
+ *
+ * @param[in]   hdr                 gnrc_netif header from whom to extract
+ * @param[out]  pointer_to_addr     pointer to address will be stored here
+ *
+ * @return                          length of destination address
+ */
+int gnrc_netif_hdr_get_dstaddr(gnrc_netif_hdr_t *hdr, uint8_t** pointer_to_addr);
```

Comment:
> For the address stuff yes.

So, for this, I return to the original state to `gnrc_mac_get_dstaddr(gnrc_pktsnip_t* pkt)` (with search functionality) and keep this function in `gnrc_mac` internal? 

>  For the flag stuff (bcast, mcast) I think it's best if they just return the flags. Otherwise you need to search for the netif header twice. The flags can then be checked externally (if necessary with a wrapper)

Fot this, you mean a solution like: `uint8_t (flag) gnrc_netif_hdr_get_flags(gnrc_netif_hdr_t *hdr)` , right? ;-)